src/modules/ganalytics/AnalyticsProvider.tsx

```diff
@@ -1,107 +1,177 @@
+      });
+      if (sendPageView) {
+        console.log('Initial google pageview');
+      }
+      setTrackingInitialized(true);
+    }
 
-    // initial config
-    configureAnalytics({
-      //   network: endpointName,
-    });
-  }, [
-    pubkey,
-    // endpointName
-  ]);
+    if (BUGSNAG_API_KEY) {
+      Bugsnag.start({
+        appVersion: '0.1.0', // TODO: Link to app version
+        apiKey: BUGSNAG_API_KEY,
+        plugins: [new BugsnagPluginReact()],
+        onError(event) {
+          if (pubkey) {
+            event.setUser(pubkey);
+          }
+          if (false) {
+            // for later inspiration
+            event.addMetadata('company', {
+              name: 'Acme Co.',
+              country: 'uk',
+            });
+          }
+        },
+      });
+    }
+  }
 
-  function setUserProperties(attributes: AnalyticsUserProperties) {
-    gtag('set', 'user_properties', {
-      ...attributes,
-    });
+  function identify() {
+    console.log(pubkey ? 'User identified' : 'No user identified');
+    if (pubkey) {
+      gtag('set', 'user_properties', {
+        user_id: pubkey,
+        pubkey: pubkey,
+      });
+    }
   }
 
-  function configureAnalytics(options: Partial<CustomEventDimensions>) {
-    if (!gtag) return;
-    gtag('config', GOOGLE_ANALYTICS_ID, {
-      ...options,
-      send_page_view: false,
+  function resetTracking() {
+    gtag('set', 'user_properties', {
+      user_id: '',
+      pubkey: '',
     });
   }
 
   function pageview(path: string) {
-    // Use this only for virtual pageviews, regular ones we get from
-    //  GA4 Enhanced page tracking
-    if (!gtag) return;
     track('page_view', {
-      path,
+      page_path: path,
     });
+    // ga4Event('page_view', {
+    //   page_path: path,
+    // });
   }
 
-  function track(
-    action: string,
-    attributes: {
-      category?: string;
-      label?: string;
-      value?: number;
-      sol_value?: number;
-      [key: string]: string | number | undefined;
-    } & Partial<CustomEventDimensions> = {}
-  ) {
-    if (!gtag) return;
-    const { category, label, sol_value, value, ...otherAttributes } = attributes;
-    gtag('event', action, {
+  // initialize (goes first no matter what)
+  useEffect(() => {
+    if (trackingAccepted) {
+      initializeTracking();
+    } else {
+      resetTracking();
+    }
+  }, [trackingAccepted]);
+
+  useEffect(() => {
+    identify();
+  }, [pubkey]);
+
+  // don't entiery trust google to track route changes
+  // PS: This does not conflict with the event listner in _app because it is a different function
+  useEffect(() => {
+    router.events.on('routeChangeComplete', pageview);
+
+    return () => {
+      router.events.off('routeChangeComplete', pageview);
+    };
+  }, [router.events]);
+
+  function track(action: AnalyticsAction, attributes: TrackingAttributes) {
+    const { category, label, value, sol_value, ...otherAttributes } = attributes;
+
+    const attrs = {
```

Comment:
Mentioned this before but the track function is becoming bloated. It should only have the attributes that are going to be on every event. And then in the page handler decorate track with specifics for that page and use the decorated track event when it comes time to fire an event.